app/models/room.rb

```diff
@@ -1,3 +1,13 @@
 class Room < ApplicationRecord
    belongs_to :hotel
+   has_many :guest_rooms, dependent: :destroy
+   has_many :guests, through: :guest_rooms
+
+   def hotel_name
+      Hotel.find(hotel_id).name
+  end
+
+  def guest_count
+    guests.distinct.count
+  end
 end
```

Comment:
In your `hotel_name` method, I wonder why you needed to find the hotel. We already have that association set up, `belongs_to :hotel`. So our logic could have just been, `hotel.name`.

Beyond that, you don't need to necessarily create a new method to get the name of a hotel because this logic is not making a call to the database, it's just retrieving the name attribute on the hotel object. You could have just called `room.hotel.name` from the view with no issue.